### PR TITLE
TIME_MILLIS support to GenericParquetReader (issue #502)

### PR DESCRIPTION
**Pull request by pranaydharmale**
added support for TIME_MILLIS to GenericParquetReader.

### CODE REVIEW

**Comment by rdblue** [comment]
Could you add a test for this?

**Comment by pranaydharmale** [comment]
> Could you add a test for this?

done; pls check out

File: data/src/test/java/org/apache/iceberg/data/DataTest.java [hunk 1]
```diff
@@ -56,7 +56,8 @@
       required(114, "dec_9_0", Types.DecimalType.of(9, 0)),
       required(115, "dec_11_2", Types.DecimalType.of(11, 2)),
       required(116, "dec_38_10", Types.DecimalType.of(38, 10)), // maximum precision
-      required(117, "time", Types.TimeType.get())
+      required(117, "time", Types.TimeType.get()),
+      required(118, "time_ms",  Types.TimeType.get())
```
Comment (rdblue) [inline data/src/test/java/org/apache/iceberg/data/DataTest.java:60]: Iceberg won't actually write data in milliseconds, so we would need to write a Parquet file using the annotation and confirm that a `time` column is correct when reading.

Can you remove this change?

Comment (pranaydharmale) [reply data/src/test/java/org/apache/iceberg/data/DataTest.java]: done.

Comment (pranaydharmale) [reply data/src/test/java/org/apache/iceberg/data/DataTest.java]: @rdblue do we have any boilerplate code I can use to do that?

**Comment by rdblue** [comment]
We don't have anything currently, but you can use `parquet-avro` classes to write a file. Let's do this in a follow-up.

**Comment by rdblue** [comment]
@pranaydharmale, I went ahead and merged this since it looks correct. We can follow up with that test.

**Pull request merged**